### PR TITLE
Improve the document of `BruteForceSampler`

### PR DESCRIPTION
**Pull request by not522**
## Motivation
This PR arranges the document of `BruteForceSampler`.

## Description of the changes
- Clarify what `BruteForceSampler` is assuming.

### CODE REVIEW

**Comment by c-bata** [comment]
@HideakiImamura Could you review this PR? You can merge this PR with only your one approval.

File: optuna/samplers/_brute_force.py [hunk 1]
```diff
@@ -25,16 +25,14 @@ class BruteForceSampler(BaseSampler):
...
     Note:
         The defined search space must be finite. Therefore, when using
-        :class:`~optuna.distributions.FloatDistibution`, `step=None` is not allowed.
+        :class:`~optuna.distributions.FloatDistibution`, ``step=None`` is not allowed.
 
     Note:
-        This sampler should not be used in combination with other samplers. For example,
-        in distributed optimization, if :class:`~optuna.samplers.BruteForceSampler` and another
-        sampler run optimizations simultaneously, complete exhaustive search may not be performed.
+        This sampler assumes that it suggests all parameters and that the search space is fixed.
```
Comment (toshihikoyanase) [inline optuna/samplers/_brute_force.py:31]: I felt this sentence was a bit ambiguous. Users might read it as `all values of parameters without context. How about specifying `combinations` explicitly?


```suggestion
        This sampler assumes that it suggests all combinations of parameters and that the search space is fixed.
```

File: optuna/samplers/_brute_force.py [hunk 2]
```diff
@@ -25,16 +25,14 @@ class BruteForceSampler(BaseSampler):
...
+        For example, the sampler may fail to try the entire search space in the following cases.
 
-    Note:
-        The objective function must be fixed during the search. Otherwise, sampler may fail to
-        try the entire search space.
+        * Using with other samplers or :meth:`~optuna.study.Study.enqueue_trial`
+        * Changing suggestions range or adding parameters in the same :class:`~optuna.study.Study`
```
Comment (toshihikoyanase) [inline optuna/samplers/_brute_force.py:35]: A nit.
```suggestion
        * Changing suggestion ranges or adding parameters in the same :class:`~optuna.study.Study`
```

**Review by toshihikoyanase — COMMENTED** [review]
I have two small comments.

**Review by HideakiImamura — APPROVED** [review]
LGTM except @toshihikoyanase's comment.

File: optuna/samplers/_brute_force.py (hunk 1 above)
Comment (not522) [reply optuna/samplers/_brute_force.py:31]: Here, I want to say that any parameters are not suggested by other features (other samplers / `enqueue_trial`).

Comment (toshihikoyanase) [reply optuna/samplers/_brute_force.py:31]: Ah, I misunderstood. Thank you for your explanation.

**Review by toshihikoyanase — APPROVED** [review]
LGTM! Thank you.

**Pull request merged**